### PR TITLE
chore(ci): Add all common release configuration from alloydb java

### PR DESCRIPTION
**Pull request by hessjcg**
This copies the standard AlloyDB build configuration files with minor changes to support 
this specific repo.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
```
Comment (enocom) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: I've been meaning to delete this file elsewhere. Let's not add it back here. Customers rarely use it correctly and we end up having to delete it more than anything.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    schedule:
+      interval: "daily"
+    # Disable version updates for Maven dependencies
+    open-pull-requests-limit: 0
+  - package-ecosystem: "pip"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    # Disable version updates for pip dependencies
+    open-pull-requests-limit: 0
```
Comment (enocom) [inline .github/dependabot.yml:14]: nit: missing newline

File: .github/sync-repo-settings.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  requiresCodeOwnerReviews: true
+  # Require up to date branches
+  requiresStrictStatusChecks: true
+  # List of required status check contexts that must pass for commits to be accepted to matching branches.
+  requiredStatusCheckContexts:
+  - "dependencies (11)"
+  - "lint"
+  - "units (8)"
+  - "units (11)"
+    #  - "graalvm17"
```
Comment (enocom) [inline .github/sync-repo-settings.yaml:48]: I think we can keep this in the repo here.

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+    - uses: actions/setup-java@5ffc13f4174014e2d4d4572b3d74c3fa61aeb2c2 # v3.11.1
+      with:
+        distribution: zulu
+        java-version: 8
+    - run: java -version
+    - run: .kokoro/build.sh
+      env:
+        JOB_TYPE: clirr
+
+          #  graalvm17:
```
Comment (enocom) [inline .github/workflows/ci.yaml:110]: Same here. This should stay I believe.

File: .kokoro/release/snapshot.cfg [hunk 5]
```diff
@@ -2,5 +2,5 @@
 
 env_vars: {
   key: "TRAMPOLINE_BUILD_FILE"
-  value: "github/cloud-sql-jdbc-socket-factory/.kokoro/release/snapshot.sh"
-}
+  value: "github/alloydb-java-connector/.kokoro/release/snapshot.sh"
+}
```
Comment (enocom) [inline .kokoro/release/snapshot.cfg:6]: nit: newline

File: .kokoro/release/snapshot.cfg [hunk 6]
```diff
@@ -2,5 +2,5 @@
 
 env_vars: {
   key: "TRAMPOLINE_BUILD_FILE"
-  value: "github/cloud-sql-jdbc-socket-factory/.kokoro/release/snapshot.sh"
-}
+  value: "github/alloydb-java-connector/.kokoro/release/snapshot.sh"
```
Comment (enocom) [inline .kokoro/release/snapshot.cfg:5]: this should be cloud-sql-jdbc-socket-factory

File: .kokoro/release/stage.cfg [hunk 7]
```diff
@@ -2,22 +2,18 @@
 
 env_vars: {
   key: "TRAMPOLINE_BUILD_FILE"
-  value: "github/cloud-sql-jdbc-socket-factory/.kokoro/release/stage.sh"
+  value: "github/alloydb-java-connector/.kokoro/release/stage.sh"
```
Comment (enocom) [inline .kokoro/release/stage.cfg:5]: Same here and below.

File: .kokoro/requirements.in [hunk 8]
```diff
@@ -0,0 +1,6 @@
+gcp-docuploader
+gcp-releasetool>=1.10.5 # required for compatibility with cryptography>=39.x
+wheel
+setuptools
+typing-extensions
+click<8.1.0
```
Comment (enocom) [inline .kokoro/requirements.in:6]: nit: newline

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,171 +1,28 @@
...
-
-Instances of abusive, harassing, or otherwise unacceptable behavior may be
-reported by opening an issue or contacting one or more of the project
-maintainers.
-
-This Code of Conduct is adapted from the [Contributor
-Covenant](http://contributor-covenant.org), version 1.2.0, available at
-[http://contributor-covenant.org/version/1/2/0/](http://contributor-covenant.org/version/1/2/0/)
-
-### Running tests
```
Comment (enocom) [inline CONTRIBUTING.md:149]: Should we keep this section?

File: samples/snapshot/pom.xml [hunk 10]
```diff
@@ -0,0 +1,83 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>-snapshot</artifactId>
+  <packaging>jar</packaging>
+  <name>Google Google Cloud AlloyDB Java Connector Snapshot Samples</name>
```
Comment (enocom) [inline samples/snapshot/pom.xml:7]: Might want to grep for AlloyDB and alloydb just to get all these

File: CONTRIBUTING.md (hunk 9 above)
Comment (hessjcg) [reply CONTRIBUTING.md:149]: I've created a more thoughful merge of CONTRIBUTING.md.

File: samples/snapshot/pom.xml (hunk 10 above)
Comment (hessjcg) [reply samples/snapshot/pom.xml]: Fixed. (I removed all the samples/ files for now.)

File: .kokoro/requirements.in (hunk 8 above)
Comment (hessjcg) [reply .kokoro/requirements.in]: fixed.

File: .kokoro/release/stage.cfg (hunk 7 above)
Comment (hessjcg) [reply .kokoro/release/stage.cfg]: Fixed.

File: .kokoro/release/snapshot.cfg (hunk 6 above)
Comment (hessjcg) [reply .kokoro/release/snapshot.cfg]: Fixed.

**Review by enocom — APPROVED** [review]

**Pull request merged**